RELEASE.md

```diff
@@ -18,6 +18,7 @@
 ## Bug fixes and other changes
 * `OmegaConfigLoader` will return a `dict` instead of `DictConfig`.
 * `OmegaConfigLoader` does not show a `MissingConfigError` when the config files exist but are empty.
+* Added `metadata` attributes to `kedro.io` datasets.
```

Comment:
This seems to be in the notes twice. I think we can just keep it under `major features and improvements`

Reply:
Should we mention the collaborative experiment tracking docs that have been added? That might draw some more attention to the feature.

Reply:
Added a section about documentation.

Reply:
Thank you @stichbury !!